contrib/Dockerfile.builder

```diff
@@ -51,4 +51,19 @@ RUN cd /tmp/ && \
     rm -rf bitcoin.tar.gz /tmp/bitcoin-$BITCOIN_VERSION
 
 RUN pip3 install --upgrade pip && \
-    python3 -m pip install python-bitcoinlib==0.7.0 pytest==3.0.5 setuptools==36.6.0 pytest-test-groups==1.0.3 flake8==3.5.0 pytest-rerunfailures==3.1 ephemeral-port-reserve==1.1.0 pytest-xdist==1.22.2 flaky==3.4.0 CherryPy==17.3.0 Flask==1.0.2
+    python3 -m pip install \
```

Comment:
Is there a reason these requirements are specifically listed out for each Dockerfile rather than just sharing 
a requirements.txt and installing like:
```
COPY requirements.txt /tmp
...
python3 -m pip install -r /tmp/requirements.txt
```

Reply:
If we want to have files that are copied in we need to send the docker build context to the docker daemon. By making the `Dockerfile`s self-contained we can use the following:

```
docker build - < contrib/Dockerfile.builder
```

Which saves us from sending the entire repo, and whatever files you have in the same directory to the daemon (2GB in my case...). Also we'd need to keep a separate `requirements.txt` since we want to add more dependencies to the builder than what we require from developers to have on their box.

Reply:
Got it :+1:  thanks for the explanation.